onc-healthit/lantern-back-end
# Validation structure

This ticket included researching the conformance criteria we want to validate as well as creating the structure for saving validation information, and that information can be found here (the structure that was implement can be found at the bottom of the file): https://mitre.sharepoint.com/:w:/r/sites/ProjectLantern/_layouts/15/Doc.aspx?sourcedoc=%7BBA06EEC7-CAAB-484E-B6CF-92FB82531A43%7D&file=validation_checks.docx&action=default&mobileredirect=true

To test:
1. Make sure there are endpoints in the database (e.g. run `make populatedb`)
2. In the `endpointmanager/cmd` directory, run `go run sendendpoints/main.go`
3. In another terminal in the same directory, run `go run capabilityreceiver/main.go`
4. Access the database and check that the validation field has been filled for every row in the `fhir_endpoints_info` table.
5. Check to make sure that validation information is being populated correctly by checking known broken values, like `select validation, http_response, mime_types from fhir_endpoints_info where http_response != 200;`

--------

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: https://oncprojectracking.healthit.gov/support/browse/LANTERN-200
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code accomplishes the tasks purpose.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.
